### PR TITLE
Add GitHub actions

### LINKED ISSUE #172: Move from Azure Pipelines to GitHub Actions
Any interest in moving to GitHub Actions? I noticed the Actions tab is turned on in this repo but isn't being used. I'm willing to do the work to set it up if you're interested.

### PR DESCRIPTION
**Pull request by brettcannon**
## Description (e.g. "Related to ...", etc.)

Introduce GitHub Actions for CI. This will run all tests and linting on each PR and push to the repository.

It will also build the example JSON extension as appropriate. This required introducing an ESLint configuration file which checks for errors and common mistakes.

Closes #172 

## Code review checklist (for code reviewer to complete)

- [X] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [X] Title summarizes what is changing
- [X] Commit messages are meaningful (see [this][commit messages] for details)
- [X] Tests have been included and/or updated, as appropriate
- [X] ~Docstrings have been included and/or updated, as appropriate~
- [X] ~Standalone docs have been updated accordingly~
- [X] [CONTRIBUTORS.md][contributors] was updated, as appropriate
- [X] ~Changelog has been updated, as needed (see [CHANGELOG.md][changelog])~

[changelog]: https://github.com/openlawlibrary/pygls/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/
[contributors]: https://github.com/openlawlibrary/pygls/blob/master/CONTRIBUTORS.md

### CODE REVIEW

**Comment by brettcannon** [comment]
Demo of the workflows running: https://github.com/brettcannon/pygls/actions.

**Comment by brettcannon** [comment]
I also didn't remove the Azure Pipelines configuration as I wasn't sure if you wanted to keep both, remove it in this PR, remove it in a separate PR, etc.

**Review by danixeee — COMMENTED** [review]
Looks good overall.

I noticed that the `json-extension` vsix artifact is not uploaded after successful [build](https://github.com/brettcannon/pygls/actions/runs/800921989). Could you check it?

```
No files were found with the provided path: *.vsix. No artifacts will be uploaded.
```

File: .github/workflows/json-extension.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            version=$(python -m json.tool package.json | awk -F'"' '/version/{print $4}')
+            build_version=$version+$2
+            build_version_line=${version_line/$version/$build_version}
+            sed -i "s|$version_line|$build_version_line|g" $1
+          }
+
+          replace_packagejson_version package.json $GITHUB_RUN_ID
+          ./node_modules/.bin/vsce package
+
+      - name: Check VSIX
```
Comment (danixeee) [inline .github/workflows/json-extension.yml:62]: You can rename this step to "Build VSIX" and move `./node_modules/.bin/vsce package` from line 60. to line 64.

**Comment by brettcannon** [comment]
I addressed the review comments and made it an error if no extension gets uploaded.

**Review by danixeee — APPROVED** [review]
Looks good now, I'll remove Azure Pipelines. Thanks!

**Pull request merged**